libs/hmpb/src/render_ballot.tsx

```diff
@@ -541,25 +541,24 @@ export async function renderAllBallotsAndCreateElectionDefinition<
   const { election } = ballotProps[0];
   assert(ballotProps.every((props) => props.election === election));
 
-  const ballotsWithLayouts = await Promise.all(
-    ballotProps.map(async (props) => {
-      // We currently only need to return errors to the user in ballot preview -
-      // we assume the ballot was proofed by the time this function is called.
-      const document = (
-        await renderBallotTemplate(renderer, template, props)
-      ).unsafeUnwrap();
-      const gridLayout = await extractGridLayout(
-        document,
-        props.ballotStyleId,
-        template
-      );
-      return {
-        document,
-        gridLayout,
-        props,
-      };
-    })
-  );
+  const ballotsWithLayouts = [];
+  for (const props of ballotProps) {
+    // We currently only need to return errors to the user in ballot preview -
+    // we assume the ballot was proofed by the time this function is called.
+    const document = (
+      await renderBallotTemplate(renderer, template, props)
+    ).unsafeUnwrap();
+    const gridLayout = await extractGridLayout(
+      document,
+      props.ballotStyleId,
+      template
+    );
+    ballotsWithLayouts.push({
+      document,
+      gridLayout,
+      props,
+    });
+  }
```

Comment:
If we want to allow but limit concurrency, another simple option could be to use [Bluebird's](http://bluebirdjs.com/docs/api/promise.map.html) `Promise.map` which takes in a `concurrency` param. But we may not want to introduce another dep

Reply:
Good point - a previous version of this PR used `p-limit` to similar effect, but @arsalansufi and I found in testing that no concurrency was more stable than limited concurrency.